discovery-provider/src/challenges/track_upload_challenge.py

```diff
@@ -0,0 +1,62 @@
+from collections import Counter
+from typing import List, Optional
+from sqlalchemy.orm.session import Session
+from src.models import Track, UserChallenge
+from src.challenges.challenge import (
+    ChallengeManager,
+    ChallengeUpdater,
+)
+
+
+class TrackUploadChallengeUpdater(ChallengeUpdater):
+    """Updates a track upload challenge.
+    Requires user to upload N (e.g. 3) songs to Audius to complete the challenge.
+    """
+
+    def update_user_challenges(
+        self,
+        session: Session,
+        event: str,
+        user_challenges: List[UserChallenge],
+        step_count: Optional[int],
+        starting_block: Optional[int],
+    ):
+        if not starting_block:
+            raise Exception("Expected a starting block for track upload challenge")
+
+        num_tracks_per_user = self._get_num_track_uploads_by_user(
+            session,
+            user_challenges,
+            starting_block,
```

Comment:
Do we need to have all this logic with the starting block, and checking the block# in the tracks table later on? The base ChallengeManager.process() method will ensure that if an event is processed prior to the challenge's start block, it gets dropped - so not sure that we need this?